azkaban-common/src/main/java/azkaban/storage/LocalStorage.java

```diff
@@ -17,76 +17,76 @@
 
 package azkaban.storage;
 
+
+import static azkaban.utils.StorageUtils.*;
 import static com.google.common.base.Preconditions.checkArgument;
 
 import azkaban.AzkabanCommonModuleConfig;
+import azkaban.spi.Dependency;
 import azkaban.spi.Storage;
 import azkaban.spi.StorageException;
-import azkaban.spi.StorageMetadata;
+import azkaban.spi.ProjectStorageMetadata;
 import azkaban.utils.FileIOUtils;
 import javax.inject.Inject;
 import javax.inject.Singleton;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import org.apache.commons.codec.binary.Hex;
 import org.apache.commons.io.FileUtils;
 import org.apache.log4j.Logger;
 
-
 @Singleton
 public class LocalStorage implements Storage {
-
   private static final Logger log = Logger.getLogger(LocalStorage.class);
 
   final File rootDirectory;
 
   @Inject
   public LocalStorage(final AzkabanCommonModuleConfig config) {
-    this.rootDirectory = validateRootDirectory(
-        createIfDoesNotExist(config.getLocalStorageBaseDirPath()));
+    this.rootDirectory = validateDirectory(createIfDoesNotExist(config.getLocalStorageBaseDirPath()));
   }
 
-  private static File createIfDoesNotExist(final String baseDirectoryPath) {
-    final File baseDirectory = new File(baseDirectoryPath);
+  private static File createIfDoesNotExist(final File baseDirectory) {
```

Comment:
this is only called once at line 58. What is the motive of making this additional function?

Reply:
Readability - and it could potentially be used by other methods in the future.